### PR TITLE
Update create sink, add Kafka sink guide

### LINKED ISSUE #550: Document SSL/SASL support for Kafka sink and some additional parameters
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/7540

### Which part(s) of the docs might be affected or should be updated? And how?

At this moment, we have only one topic about sink (`CREATE SINK`). Similar to creating sources, we need to have separate topics for different sinks (currently Kafka and JDBC). These should be at least two topics. This is because there will be different sets of parameters for different connectors, and holding them all in one topic is hard for people to find.

To be clear, I think we need the `CREATE SINK` topic to be as generic as possible and serves as the landing page and points to the two guides that we are about to create.

After we have these two topics, we can update the Kafka sink topic by adding these SSL/SASL parameters (or simply referring to the Kafka source topic).

### Reference

https://www.risingwave.dev/docs/upcoming/create-source-kafka/

### LINKED ISSUE #581: Document force_append_only parameter for creating a sink
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/7922

### Which part(s) of the docs might be affected or should be updated? And how?

When this option is set to `true`, even if the sink-from object is not append-only, the sink can be converted to be append-only. Please see the comments in the code PR for details.

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
## Info
- **Description**: 
Simplify CREATE SINK SQL reference page
Add Kafka sink guide
Remove CREATE SINK from data delivery, add Kafka sink guide
Document force_append_only and SSL/SASL support (updates found in Kafka sink guide) 

- **Preview**: 
Kafka Sink: https://pr-588.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/create-sink-kafka/
Create Sink: https://pr-588.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/sql-create-sink/

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/7540
https://github.com/risingwavelabs/risingwave/pull/7922

- **Related doc issue**: 
https://github.com/risingwavelabs/risingwave-docs/issues/550
https://github.com/risingwavelabs/risingwave-docs/issues/581

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/guides/create-sink-kafka.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+```sql
+CREATE SINK [ IF NOT EXISTS ] sink_name
+[FROM sink_from | AS select_query]
+WITH (
+   connector='kafka',
+   field_nake = 'value', ...
+);
+```
+
+## Parameters
```
Comment (neverchanje) [inline docs/guides/create-sink-kafka.md]: Basic Parameters. As a reader I might feel if security is unsupported

**Review by neverchanje — APPROVED** [review]
rest lgtm

File: docs/sql/commands/sql-create-sink.md [hunk 2]
```diff
@@ -3,23 +3,19 @@ id: sql-create-sink
...
 ```sql
 CREATE SINK [ IF NOT EXISTS ] sink_name
 [FROM sink_from | AS select_query]
 WITH (
-   connector='kafka',
-   kafka.brokers='broker_address',
-   kafka.topic='topic_address',
-   format='format'
+   connector='connector_name',
+   field_nake = 'value', ...
```
Comment (hengm3467) [inline docs/sql/commands/sql-create-sink.md]: Yesterday we had a PR that changed `field_name` to `connector_parameter`. Let's use the latter as the way to refer to these parameters. Fields might not very straightforward for some.

File: sidebars.js [hunk 3]
```diff
@@ -216,11 +216,10 @@ const sidebars = {
       },
       {
         type: 'doc',
-        id: 'sql/commands/sql-create-sink',
-        label: 'Create a sink',
-      },
-    ]
-  },
+        label: 'Create Kafka sink',
```
Comment (hengm3467) [inline sidebars.js]: @emile-00  To be consistent with "Data ingestion", let's use just "Kafka" as the label of the topic.

File: docs/guides/create-sink-kafka.md [hunk 4]
```diff
@@ -0,0 +1,322 @@
+---
+id: create-sink-kafka
+title: SINK TO KAFKA
```
Comment (hengm3467) [inline docs/guides/create-sink-kafka.md]: No need to use all caps for the title. I'd suggest we change it to "Sink data to Kafka". @emile-00

File: docs/guides/create-sink-kafka.md [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+description: Sink to Kafka.
```
Comment (hengm3467) [inline docs/guides/create-sink-kafka.md]: I know we have been adding descriptions the same as titles, but we need to provide a bit more info. For example: Sink data in RisingWave to Kafka topics.

**Review by hengm3467 — APPROVED** [review]
Otherwise looks good to me. Really solid work! Thanks. @emile-00

**Review by tabVersion — APPROVED** [review]

File: docs/guides/create-sink-kafka.md [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+|AS select_query| A SELECT query that specifies the data to be output to the sink. Either this query or a FROM clause must be specified. See [SELECT](../commands/sql-select.md) for the syntax and examples of the SELECT command.|
+|connector| Sink connector type. Currently, only `‘kafka’` and `‘jdbc’` are supported. If there is a particular sink you are interested in, see the [Integrations Overview](../../rw-integration-summary.md) page for a full list of connectors and integrations we are working on. |
+|properties.bootstrap.server|Address of the Kafka broker. Format: `‘ip:port’`. If there are multiple brokers, separate them with commas. |
+|topic|Address of the Kafka topic. One sink can only correspond to one topic.|
+|format|Data format. Allowed formats:<ul><li> `append_only`: Output data with insert operations.</li><li> `debezium`: Output change data capture (CDC) log in Debezium format.</li></ul>|
+|force_append_only| If `true`, forces the sink to be `append_only`, even if it cannot be.| 
+|use_transaction| |
```
Comment (hengm3467) [inline docs/guides/create-sink-kafka.md:37]: @emile-00  I think `use_transaction` is not explained.

**Pull request merged**